sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/LeftSemiAntiJoinPushDownSuite.scala

```diff
@@ -314,4 +315,116 @@ class LeftSemiPushdownSuite extends PlanTest {
     val optimized = Optimize.execute(originalQuery.analyze)
     comparePlans(optimized, originalQuery.analyze)
   }
+
+  Seq(LeftSemi, LeftAnti).foreach { case outerJT =>
+    Seq(Inner, LeftOuter, Cross, RightOuter).foreach { case innerJT =>
+      test(s"$outerJT pushdown with empty join condition join type $innerJT") {
+        val joinedRelation = testRelation1.join(testRelation2, joinType = innerJT, None)
+        val originalQuery = joinedRelation.join(testRelation, joinType = outerJT, None)
+        val optimized = Optimize.execute(originalQuery.analyze)
+
+        val correctAnswer = if (innerJT == RightOuter) {
+          val pushedDownJoin = testRelation2.join(testRelation, joinType = outerJT, None)
+          testRelation1.join(pushedDownJoin, joinType = innerJT, None)
+        } else {
+          val pushedDownJoin = testRelation1.join(testRelation, joinType = outerJT, None)
+          pushedDownJoin.join(testRelation2, joinType = innerJT, None)
+        }
+        comparePlans(optimized, correctAnswer)
+      }
+    }
+  }
+
+  Seq(Some('d === 'e), None).foreach { case innerJoinCond =>
+    Seq(LeftSemi, LeftAnti).foreach { case outerJT =>
+      Seq(Inner, LeftOuter, Cross).foreach { case innerJT =>
```

Comment:
These test cases almost cover the above one, except the RightOuter join type. Can we merge them.

Reply:
nvm, I was wrong